chips/stm32f4xx/src/chip_specific.rs

```diff
@@ -0,0 +1,269 @@
+        Latency6,
+        /// 7 wait cycles
+        Latency7,
+        /// 8 wait cycles
+        Latency8,
+        /// 9 wait cycles
+        Latency9,
+        /// 10 wait cycles
+        Latency10,
+        /// 11 wait cycles
+        Latency11,
+        /// 12 wait cycles
+        Latency12,
+        /// 13 wait cycles
+        Latency13,
+        /// 14 wait cycles
+        Latency14,
+        /// 15 wait cycles
+        Latency15,
+    }
+
+    #[cfg(any(
+        feature = "stm32f405",
+        feature = "stm32f415",
+        feature = "stm32f407",
+        feature = "stm32f417"
+    ))]
+    #[derive(Copy, Clone, PartialEq, Debug)]
+    /// Enum representing all the possible values for the flash latency
+    pub(crate) enum FlashLatency {
+        /// 0 wait cycles
+        Latency0,
+        /// 1 wait cycle
+        Latency1,
+        /// 2 wait cycles
+        Latency2,
+        /// 3 wait cycles
+        Latency3,
+        /// 4 wait cycles
+        Latency4,
+        /// 5 wait cycles
+        Latency5,
+        /// 6 wait cycles
+        Latency6,
+        /// 7 wait cycles
+        Latency7,
+    }
+
+    // Chip-specific trait that allows to read the current configured flash latency
+    pub(crate) trait SpecificFlashTrait {
+        // The number of wait cycles depends on two factors: system clock frequency and the supply
+        // voltage. Currently, this method assumes 2.7-3.6V voltage supply (default value).
+        // TODO: Take into the account the power supply
+        //
+        // The number of wait states varies from chip to chip.
+        fn get_number_wait_cycles_based_on_frequency(&self, frequency_mhz: usize) -> FlashLatency {
+            #[cfg(not(any(
+                feature = "stm32f410",
+                feature = "stm32f411",
+                feature = "stm32f412",
+                feature = "stm32f413",
+                feature = "stm32f423"
+            )))]
+            {
+                if frequency_mhz <= 30 {
+                    FlashLatency::Latency0
+                } else if frequency_mhz <= 60 {
+                    FlashLatency::Latency1
+                } else if frequency_mhz <= 90 {
+                    FlashLatency::Latency2
+                } else if frequency_mhz <= 120 {
+                    FlashLatency::Latency3
+                } else if frequency_mhz <= 150 {
+                    FlashLatency::Latency4
+                } else {
+                    FlashLatency::Latency5
+                }
+            }
+            #[cfg(any(feature = "stm32f410", feature = "stm32f411", feature = "stm32f412"))]
+            {
+                if frequency_mhz <= 30 {
+                    FlashLatency::Latency0
+                } else if frequency_mhz <= 64 {
+                    FlashLatency::Latency1
+                } else if frequency_mhz <= 90 {
+                    FlashLatency::Latency2
+                } else {
+                    FlashLatency::Latency3
+                }
```

Comment:
```suggestion
                match frequency_mhz {
                    0..=30 => FlashLatency::Latency0,
                    31..=64 => FlashLatency::Latency1,
                    65..=90 => FlashLatency::Latency2,
                    _ => FlashLatency::Latency3,
                }
```

I can't believe I'm asking this, but, would this be more a idiomatic way to express this in Rust?

Reply:
Yes, it is. I'm sorry for this!

Reply:
Hah, no reason to apologize — I was laughing at myself because I only ever _get_ comments like that re: idiomatic Rust. I think this is the first time I've ever been able to give one correctly. Only took me eight or so years, but I guess I finally grok the Rust ethos 😅